ci/ray_ci/tester.py

```diff
@@ -414,23 +415,63 @@ def _get_high_impact_test_targets(
         for test in itertools.chain.from_iterable(step_id_to_tests.values())
         if test.get_oncall() == team
     }
-    new_tests = _get_new_tests(os_prefix, container)
+    changed_tests = _get_changed_tests()
 
-    return high_impact_tests.union(new_tests)
+    return high_impact_tests.union(changed_tests)
 
 
-def _get_new_tests(prefix: str, container: TesterContainer) -> Set[str]:
+def _get_changed_tests() -> Set[str]:
     """
-    Get all local test targets that are not in database
+    Get all changed tests in the current PR
     """
-    local_test_targets = set(
-        container.run_script_with_output(['bazel query "tests(//...)"'])
-        .strip()
-        .split(os.linesep)
+    changed_files = _get_changed_files()
+    logger.info(f"Changed files: {changed_files}")
+    return set(
+        itertools.chain.from_iterable(
+            [_get_test_targets_per_file(file) for file in _get_changed_files()]
+        )
     )
-    db_test_targets = {test.get_target() for test in Test.gen_from_s3(prefix=prefix)}
 
-    return local_test_targets.difference(db_test_targets)
+
+def _get_test_targets_per_file(file: str) -> Set[str]:
+    """
+    Get the test target from a file path
+    """
+    try:
+        package = subprocess.check_output(["bazel", "query", file]).decode().strip()
+        if not package:
+            return set()
+        targets = subprocess.check_output(
+            ["bazel", "query", f"tests(attr('srcs', {package}, //...))"]
+        )
+        targets = {
+            target.strip()
+            for target in targets.decode().splitlines()
+            if target is not None
+        }
+        logger.info(f"Found test targets for file {file}: {targets}")
+
+        return targets
+    except subprocess.CalledProcessError:
+        logger.info(f"File {file} is not a test target")
+        return set()
+
+
+def _get_changed_files() -> Set[str]:
+    """
+    Get all changed files in the current PR
+    """
+    base = os.environ.get("BUILDKITE_PULL_REQUEST_BASE_BRANCH")
+    head = os.environ.get("BUILDKITE_COMMIT")
+    if not base or not head:
+        # if not in a PR, return an empty set
+        return set()
+
+    subprocess.check_call(["git", "fetch", "origin", base])
+    changes = subprocess.check_output(
+        ["git", "diff", "--name-only", f"origin/{base}...{head}"]
+    )
```

Comment:
you need to run this (and other bazel commands) in the right working directory? if it is run with `bazel run`, it is not running under the root?

it might still work, but probably better to make sure it is not influenced by how `bazel run` works?

Reply:
there is this line that you told me to add at some point https://github.com/ray-project/ray/blob/master/ci/ray_ci/tester.py#L205; is `bazel_workspace_dir` the same as ray checkout directory?

Reply:
back to your queue to make sure you see this; historically i always run commands in the container for the same reasons; but i just discover this line `os.chdir` recently

Reply:
yes.

you can get the repo's root via the bazel workspace dir thing, and then you can just send in `cwd=` into `check_output`.

Reply:
understood, but there is already this line https://github.com/ray-project/ray/blob/master/ci/ray_ci/tester.py#L205, so the `cwd` is not needed; or you still want the `cwd` to be safe?

Reply:
> you still want the cwd to be safe?

yes. I don't want an underlying function to have the constraint that it only works under particular cwd. can we pass the repo's root dir in on init or something?